### PR TITLE
Update documentation for `lufact!`. Fixes #16193.

### LINKED ISSUE #16193: `lufact` and `lufact!` behave differently on matrices of type `Int`
Hi, 

short demo:

```julia
julia> D = [1 2; 3 4]
2x2 Array{Int64,2}:
 1  2
 3  4

julia> lufact(D)
Base.LinAlg.LU{Float64,Array{Float64,2}}(2x2 Array{Float64,2}:
 1.0   2.0
 3.0  -2.0,Int32[1,2],0)

julia> lufact!(D)
ERROR: InexactError()
 in generic_lufact! at linalg/lu.jl:53
 in lufact! at linalg/lu.jl:20
```

While I understand why this happens, I believe a more detailed error message for `lufact!` would be good to have. Furthermore, wouldn't it be better to require explicitly that the input to `lufact` is of an adequate type, rather than the implicit convert? I guess more consistency between the copying and in-place versions would make this more predictable.

Cheers

### PR DESCRIPTION
**Pull request by gasagna**
Reference JuliaLang/LinearAlgebra.jl#327

### CODE REVIEW

**Comment by andreasnoack** [comment]
Great. Would you mind adding similar sentences to `cholfact!` and `qrfact!`? Maybe also `scale!`.

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -5671,10 +5671,12 @@ Base.(:(.!=))
...
     lufact!(A) -> LU
 
-`lufact!` is the same as [`lufact`](:func:`lufact`), but saves space by overwriting the
-input `A`, instead of creating a copy.  For sparse `A` the `nzval` field is not overwritten
-but the index fields, `colptr` and `rowval` are decremented in place, converting from
-1-based indices to 0-based indices.
+`lufact!` is the same as [`lufact`](:func:`lufact`), but saves space by overwriting the 
+input `A`, instead of creating a copy. An `InexactError` exception is thrown if the 
+factorisation produces a number not representable by the element type of `A`, e.g. for 
+integer types. For sparse `A` the `nzval` field is not overwritten but the index fields,
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: @andreasnoack didn't we get rid of sparse `lufact!` for now?

Comment (andreasnoack) [inline base/docs/helpdb/Base.jl]: Yes. There is no sparse `lufact!`.

**Comment by gasagna** [comment]
Ok. Removed trailing whitespace, added similar sentences to qrfact! and cholfact!, ran make docs and committed changes to .rst.  Should I do something else?

File: base/linalg/cholesky.jl [hunk 2]
```diff
@@ -126,7 +126,10 @@ end
 """
     cholfact!(A::StridedMatrix, uplo::Symbol, Val{false}) -> Cholesky
 
-The same as `cholfact`, but saves space by overwriting the input `A`, instead of creating a copy.
+The same as `cholfact`, but saves space by overwriting the input `A`, instead
+of creating a copy. An ``InexactError`` exception is thrown if the factorisation
+produces a number not representable by the element type of ``A``, e.g. for
```
Comment (tkelman) [inline base/linalg/cholesky.jl]: single backticks in markdown unless you want latex style output

File: base/linalg/cholesky.jl [hunk 3]
```diff
@@ -139,7 +142,10 @@ end
 """
     cholfact!(A::StridedMatrix, uplo::Symbol, Val{true}) -> PivotedCholesky
 
-The same as `cholfact`, but saves space by overwriting the input `A`, instead of creating a copy.
+The same as `cholfact`, but saves space by overwriting the input `A`, instead
+of creating a copy. An ``InexactError`` exception is thrown if the
```
Comment (tkelman) [inline base/linalg/cholesky.jl]: single backticks

**Comment by gasagna** [comment]
WOW, just gone through my first rebase. Should be ok now. Thanks for reviewing.

**Comment by andreasnoack** [comment]
LGTM. @tkelman Do you see anything that requires adjustments?

**Pull request merged**